IOMC/EventVertexGenerators/src/BaseEvtVtxGenerator.cc

```diff
@@ -47,20 +49,39 @@ void BaseEvtVtxGenerator::produce(Event& evt, const EventSetup&) {
 
   Handle<HepMCProduct> HepUnsmearedMCEvt;
 
-  evt.getByToken(sourceToken, HepUnsmearedMCEvt);
+  bool found = evt.getByToken(sourceToken, HepUnsmearedMCEvt);
+
+  if (found) {  // HepMC event exists
+
+    // Make a copy
+    HepMC::GenEvent* genevt = new HepMC::GenEvent(*HepUnsmearedMCEvt->GetEvent());
+
+    std::unique_ptr<edm::HepMCProduct> HepMCEvt(new edm::HepMCProduct(genevt));
+    // generate new vertex & apply the shift
+    //
+    HepMCEvt->applyVtxGen(newVertex(engine));
 
-  // Copy the HepMC::GenEvent
-  HepMC::GenEvent* genevt = new HepMC::GenEvent(*HepUnsmearedMCEvt->GetEvent());
-  std::unique_ptr<edm::HepMCProduct> HepMCEvt(new edm::HepMCProduct(genevt));
-  // generate new vertex & apply the shift
-  //
-  HepMCEvt->applyVtxGen(newVertex(engine));
+    //HepMCEvt->LorentzBoost( 0., 142.e-6 );
```

Comment:
@mkirsano , this commented line should be removed.

Reply:
done